WordPress/gutenberg
# Navigation: Add post format variation to navigation link block

## Description
Assists with https://github.com/WordPress/gutenberg/issues/29793.

Adds a Post Format variation to that navigation link block.

https://github.com/WordPress/gutenberg/pull/22600 added the ability to search for a post format, but it appears this was never turned into a block variation, so post formats can only be created as custom links.

This new variation needs an icon. At the moment it just shows the custom post type icon. But I think it's fine to merge like this and add the icon later.

## How has this been tested?
1. Use a theme with post formats. I use Twenty Fourteen.
2. Create a post and assign it a post format (the UI for this is underneath the publish date in the sidebar of the post editor). Remember the type of post format you used.
3. Give the post a memorable title and publish it
4. Create another post
5. Add the navigation block
6. Add a Post Format to the navigation block
7. Search for and use the post format assigned in step 2 (it will probably be shown without you needing to type anything)
8. Set this as the link
9. Preview the post
10. Click the link to the post format
11. The post formats screen should be displayed showing posts of the format you assigned.

## Screenshots 
<img width="335" alt="Screenshot 2021-03-31 at 6 50 52 pm" src="https://user-images.githubusercontent.com/677833/113133281-10195000-9252-11eb-991b-8bdc2f0a3252.png">

## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
